kernel/kheap.c

```diff
@@ -66,6 +66,9 @@ void *k_heap_aligned_alloc(struct k_heap *h, size_t align, size_t bytes,
 {
 	int64_t now, end = sys_clock_timeout_end_calc(timeout);
 	void *ret = NULL;
+
+	end = K_TIMEOUT_EQ(timeout, K_FOREVER) ? INT64_MAX : end;
```

Comment:
Do we really like the ternary here? Not a fan, "else" case being a no-op makes one wonder if that was really the intent, this could just be

```c
if (K_TIMEOUT_EQ(timeout, K_FOREVER) {
        end = INT64_MAX;
}
```

Anyway, up to you folks.

Reply:
Personally I'm a fan of vertical compression; source code gets too long to read in most modern styles.  Semantically my brain reads that like MIN/MAX/CLAMP, where the if() signals "behavior".  But these are going to generate the same code, so I don't see much reason to care.

Reply:
That being the case I prefer to leave as-is.

Reply:
Cool with that, thanks for the quick turnaround folks!